packages/FormElement/src/components/Content/Content.js

```diff
@@ -9,25 +10,144 @@ const propTypes = {
+const nativeElementsRequiringIdForLabel = ["input"];
+const componentsRequiringA11yDescribedByIds = [
+  "Input",
+  "Checkbox",
+  "Radio.Group",
+  "TimePicker",
+  "DatePicker",
+  "children",
+  "FormElement",
+];
+const componentsRequiringLabelRef = ["ListBox"];
+
+const isOtherChildRequiringId = (key, element) =>
+  key === "children" && nativeElementsRequiringIdForLabel.includes(element?.type);
+
+const cloneWithAriaAttributes = ({ element, idForLabel, a11yDescribedByIds, isNative, refLabel }) => {
+  const a11yProps = {};
+
+  if (a11yDescribedByIds && isNative) {
+    a11yProps["aria-describedby"] = a11yDescribedByIds;
+  }
+
+  if (a11yDescribedByIds && !isNative) {
+    a11yProps.a11yDescribedByIds = a11yDescribedByIds;
+  }
+
+  if (idForLabel) {
+    a11yProps.id = idForLabel;
+  }
+
+  if (refLabel) {
+    a11yProps.refLabel = refLabel;
+  }
+
+  return React.cloneElement(element, {
+    ...a11yProps,
+  });
+};
+
+const createFormElementChild = ({ element, a11yDescribedByIds }) => {
+  const formElementChildren = element.props.children.map(item =>
+    item.type.displayName === "FormElement.Content"
+      ? React.cloneElement(item, { wrapperA11yDescribedByIds: a11yDescribedByIds })
+      : item
+  );
+
+  return React.cloneElement(element, { children: formElementChildren });
 };
 
 function Content(props) {
-  const { children, idForLabel, refLabel, ariaDescribedBy, ...moreProps } = props;
+  const { children, idForLabel, refLabel, wrapperA11yDescribedByIds, ...moreProps } = props;
 
   if (!children) {
     return null;
   }
 
+  if (typeof children === "function") {
+    console.error("FormElement.Content no longer accepts a render function child, skipping rendering children");
+    return null;
+  }
+
+  let isLabelSet = false;
+
+  const a11yDescribedByIds = wrapperA11yDescribedByIds
+    ? `${props.a11yDescribedByIds} ${wrapperA11yDescribedByIds}`
+    : props.a11yDescribedByIds;
+
+  const extractedChildren = extractChildren(children, Object.values(supportedComponentNames));
+
+  const hasLabelAssociationOverride = Boolean(
+    Array.isArray(children)
+      ? children.find(element => element.props.overridelabelassociation)
+      : children.props.overridelabelassociation
+  );
+
+  const getSingleIdForLabel = (key, element) => {
+    if (hasLabelAssociationOverride) {
+      if (element.props.overridelabelassociation) {
+        if (isLabelSet) {
+          console.error("Label override via 'overridelabelassociation' should only be applied to one child");
+        }
+        isLabelSet = true;
+        return idForLabel;
+      }
+      return null;
+    }
+
+    if (!isLabelSet) {
+      const singleIdForLabel =
+        componentsRequiringIdForLabel.includes(key) || isOtherChildRequiringId(key, element) ? idForLabel : null;
+      isLabelSet = Boolean(singleIdForLabel);
+      return singleIdForLabel;
+    }
+  };
+
+  if (!extractedChildren?.children.length) {
+    delete extractedChildren.children;
```

Comment:
extractedChildrenHelper generates an empty children array when not needed, perhaps something we should prevent from happening